### PR TITLE
Use cases.Title instead of strings.Title because of deprecated

### PR DESCRIPTION
**Pull request by utam0k**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

`strings.Title()` has been duplicated since go 1.18[^1]

[^1]: https://go.dev/doc/go1.18

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/eventhandlers.go [hunk 1]
```diff
@@ -511,7 +514,7 @@ func addAllEventHandlers(
 			gvr, _ := schema.ParseResourceArg(string(gvk))
 			dynInformer := dynInformerFactory.ForResource(*gvr).Informer()
 			if handlerRegistration, err = dynInformer.AddEventHandler(
-				buildEvtResHandler(at, gvk, strings.Title(gvr.Resource)),
+				buildEvtResHandler(at, gvk, cases.Title(language.Und, cases.NoLower).String(gvr.Resource)),
```
Comment (utam0k) [inline pkg/scheduler/eventhandlers.go:517]: `cases.NoLower` is required to match the behavior with `strings.Title`, but if you don't need it, please let me know.

**Comment by utam0k** [comment]
> /sig architecture

@AxeZhan Thanks for your triage🙏Do we also need to add the sig-schedule?

**Comment by AxeZhan** [comment]
> Do we also need to add the sig-schedule?

Yes, as this change involves files belong to sig-scheduling.
I found that `strings.Title` is not only used in scheduler, but also api-server and many other places. So I tend to let folks from architecture to decide if we should add the dependency and replace them.

**Pull request closed without merging**